internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,393 @@
+)
+
+type initProc struct {
+	initDoOnce sync.Once
+	proc       *JobProcess
+	initBlock  chan struct{}
+}
+
+// JobContainer represents a lightweight container composed from a job object.
+type JobContainer struct {
+	id             string
+	spec           *specs.Spec          // OCI spec used to create the container
+	job            *jobobject.JobObject // Object representing the job object the container owns
+	sandboxMount   string               // path to where the sandbox is mounted on the host
+	closedWaitOnce sync.Once
+	init           initProc
+	exited         chan struct{}
+	waitBlock      chan struct{}
+	waitError      error
+}
+
+var _ cow.ProcessHost = &JobContainer{}
+var _ cow.Container = &JobContainer{}
+
+func newJobContainer(id string, s *specs.Spec) *JobContainer {
+	return &JobContainer{
+		id:        id,
+		spec:      s,
+		waitBlock: make(chan struct{}),
+		init:      initProc{initBlock: make(chan struct{})},
+	}
+}
+
+// Create creates a new JobContainer from `s`.
+func Create(ctx context.Context, id string, s *specs.Spec) (_ cow.Container, err error) {
+	log.G(ctx).WithField("id", id).Debug("Creating job container")
+
+	if s == nil {
+		return nil, errors.New("Spec must be supplied")
+	}
+
+	if id == "" {
+		g, err := guid.NewV4()
+		if err != nil {
+			return nil, err
+		}
+		id = g.String()
+	}
+
+	if err := mountLayers(ctx, s); err != nil {
+		return nil, errors.Wrap(err, "failed to mount container layers")
+	}
+
+	const volumeGUIDRegex = `^\\\\\?\\(Volume)\{{0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}\}(|\\)$`
+	if matched, err := regexp.MatchString(volumeGUIDRegex, s.Root.Path); !matched || err != nil {
+		return nil, fmt.Errorf(`invalid container spec - Root.Path '%s' must be a volume GUID path in the format '\\?\Volume{GUID}\'`, s.Root.Path)
+	}
+	if s.Root.Path[len(s.Root.Path)-1] != '\\' {
+		s.Root.Path += `\` // Be nice to clients and make sure well-formed for back-compat
+	}
+
+	container := newJobContainer(id, s)
+
+	// Create the job object all processes will run in. We set true for the notifications argument
+	// as we always want to receive messages from the job.
+	job, err := jobobject.Create(ctx, id, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create job object")
+	}
+	container.job = job
+
+	var path string
+	defer func() {
+		if err != nil {
+			container.Close()
+			if path != "" {
+				removeSandboxMountPoint(ctx, path)
+			}
+		}
+	}()
+
+	limits, err := specToLimits(ctx, s)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to convert OCI spec to job object limits")
+	}
+
+	// Set resource limits on the job object based off of oci spec.
+	if err := job.SetResourceLimits(limits); err != nil {
+		return nil, errors.Wrap(err, "failed to set resource limits")
+	}
+
+	// Setup directory sandbox volume will be mounted
+	sandboxPath := fmt.Sprintf(sandboxMountFormat, id)
+	if _, err := os.Stat(sandboxPath); err != nil {
```

Comment:
is it supposed to happen on all errors or only on `os.IsNotExists`?

Reply:
Probably just `IsNotExists` would be better